### PR TITLE
fix(ESSNTL-5049): Resolve more granular RBAC issues

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address the new bugs found in the comments of [ESSNTL-5049](https://issues.redhat.com/browse/ESSNTL-5049):
- Creating groups with inventory:groups:write permission with attributeFilter now returns HTTP 403 with a more descriptive message.
- Fixed the DB-specific RBAC group ID filter so that it works when the attributeFilter contains a null (None).
- Putting a value that's neither a UUID nor None now results in error 503, because that implies the RBAC service has populated the permissions with invalid values. 

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by jpramos123 — APPROVED** [review]
LGTM!

**Review by thearifismail — APPROVED** [review]
LGTM.  There is a minor question though

File: api/group.py [hunk 1]
```diff
@@ -72,7 +72,10 @@ def create_group(body, rbac_filter=None):
     # If there is an attribute filter on the RBAC permissions,
     # the user should not be allowed to create a group.
     if rbac_filter is not None:
-        return Response(None, status.HTTP_403_FORBIDDEN)
+        return Response(
+            "Unfiltered inventory:groups:write RBAC permission is required in order to create new groups.",
```
[the remark this answers is not in the document]
Comment (kruai) [reply api/group.py:76]: Yes - if there is an attributeFilter limiting your groups:write permissions, you're not allowed to create a group.

**Review by FabriciaDinizRH — APPROVED** [review]

**Pull request merged**